src/payment/views.py

```diff
@@ -0,0 +1,154 @@
+import stripe
+from django.conf import settings
+from django.shortcuts import get_object_or_404
+from django.urls import reverse
+from rest_framework import status, viewsets
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from payment.models import StripeProduct, Subscription
+from payment.serializers import SubscriptionSerializer
+
+ESSENTIAL_PLAN_NAME = "Essentiel"
+
+
+class SubscriptionViewSet(viewsets.ModelViewSet):
+    stripe.api_key = settings.STRIPE_API_KEY
+    stripe.api_version = settings.STRIPE_API_VERSION
+    plan_name = ESSENTIAL_PLAN_NAME = "Essentiel"
+
+    queryset = Subscription.objects.all()
+    serializer_class = SubscriptionSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(user=self.request.user)
+
+    def create(self, request, *args, **kwargs):
+        user = request.user
+        plan = request.data.get("plan")  # "monthly" ou "annual"
+
+        try:
+            product = StripeProduct.objects.get(name=self.plan_name)
+            price_id = (
+                product.monthly_price_id
+                if plan == "monthly"
+                else product.annual_price_id
+            )
+        except StripeProduct.DoesNotExist:
+            return Response({"error": "Product not found"}, status=404)
```

Comment:
can't we `get_object_or_404()`?
https://docs.djangoproject.com/en/5.1/topics/http/shortcuts/#get-object-or-404